templates/ _helpers.tpl

```diff
@@ -73,3 +73,123 @@ Create the name of the service account to use
 {{ "false" }}
 {{- end -}}
 {{- end }}
+
+{{/*
+Given the $.tonicSsoConfig, create all environment variables needed for
+the deployment. This should only be called if $.tonicSsoConfig is populated
+*/}}
+{{- define "tonic.sso" -}}
+{{- $provider := (required "tonicSsoConfig.provider is required for SSO" .provider | lower) -}}
+- name: TONIC_SSO_PROVIDER
+  value: {{ quote .provider }}
+{{- if .groupFilter -}}
+- name: TONIC_SSO_GROUP_FILTER_REGEX
+  value: {{quote .groupFilter}}
+{{- end -}}
+{{- if eq $provider "aws" }}
+{{ include "tonic.sso.aws" . }}
+{{- else if eq $provider "azure" }}
+{{ include "tonic.sso.azure" . }}
+{{- else if eq $provider "duo" }}
+{{ include "tonic.sso.duo" . }}
+{{- else if eq $provider "google" }}
+{{ include "tonic.sso.google" . }}
+{{- else if eq $provider "okta" }}
+{{ include "tonic.sso.okta" . }}
+{{- else if eq $provider "keycloak" }}
+{{ include "tonic.sso.keycloak" . }}
+{{- else if eq $provider "saml" }}
+{{ include "tonic.sso.saml" . }}
+{{- else -}}
+{{- fail "Unsupported SSO provider " .provider -}}
+{{- end -}}
+{{- end -}}
+
+{{- define "tonic.sso.aws" -}}
+{{- if .samlIdpMetadataUrl -}}
+- name: TONIC_SSO_SAML_IDP_METADATA_XML_URL
+  value: {{ quote .samlIdpMetadataUrl }}
+{{- else if .samlIdpMetadataXml }}
+- name: TONIC_SSO_SAML_IDP_METADATA_XML_BASE64
+  value: {{ quote .samlIdpMetadataXml }}
+{{- else -}}
+{{ fail "Either samlIdpMetadataUrl or samlIdpMetadataXml must be provided to configre AWS SSO" }}
+{{- end -}}
+{{- end -}}
+
+{{- define "tonic.sso.azure" -}}
+- name: TONIC_SSO_TENANT_ID
+  value: {{ quote .tenantId }}
```

Comment:
I'm going to mark all non-conditional values with `required "blahblahblah is required to configure <provider> sso"`, for now these were extracted raw from the existing deployment template.